Simulator Tests in CI

gijskoning opened the pull request:
Edit:
Add 3 tests to verify bridge.py and the all processes of manager.py correctly startup and keep running for 60 seconds. Each need Carla simulator to run in parallel.
First two test simply verify that connection to Carla is possible and that bridge.py works as expected.
The third runs manager and bridge. Checkes processes are running and that car can be engaged.

Bridge.py looks like a lot of changes but it is mostly creating a Bridge class and putting some functionality in functions

Also speeds up simulation by changing post processes to false

gijskoning committed:
Add c3 camera flag to bridge.py

gijskoning committed:
Set c3_cameras to default

gijskoning committed:
add runtime error message

gijskoning committed:
Add CI test for connecting bridge.py with running carla container

gijskoning committed:
Add CI test for connecting bridge.py with running carla container

gijskoning committed:
Add CI test for connecting bridge.py with running carla container

gijskoning committed:
Add CI test for connecting bridge.py with running carla container

gijskoning committed:
Fix

gijskoning committed:
Fix

gijskoning committed:
Undo removal plotjuggler
Fix build

gijskoning committed:
Undo removal plotjuggler
Fix build

gijskoning committed:
undo removal plotjuggler

gijskoning committed:
undo removal plotjuggler

adeebshihadeh commented:
For this PR, let's focus on just writing a test and remove the CI stuff. We'll probably need to run this on our Jenkins machine with a GPU, which I'll setup after this is merged.

gijskoning committed:
Remove CI stuff.
Add basic carla bridge test

gijskoning committed:
Remove CI stuff.
Add basic carla bridge test

gijskoning committed:
Remove CI stuff.
Add basic carla bridge test

gijskoning committed:
Remove CI stuff.
Add basic carla bridge test

gijskoning committed:
Small change

gijskoning committed:
Small change

gijskoning committed:
Small fix

gijskoning committed:
Small fix

gijskoning commented:
Thanks for your feedback. Removed the CI stuff. 
Created basic unittests for connecting with Carla without errors and another for running the bridge without errors for 50 seconds.

I moved some functionality in a bridge class which makes it a bit easier to test.

gijskoning committed:
Final fixes

gijskoning committed:
Final fixes

gijskoning committed:
Final fixes

gijskoning committed:
Final fixes

gijskoning committed:
Small change

gijskoning committed:
Small change

gijskoning committed:
Merge branch 'master' into carla-bridge-ci

# Conflicts:
#	tools/sim/bridge.py

gijskoning committed:
Merge branch 'master' into carla-bridge-ci

gijskoning committed:
Add manager integration test

gijskoning committed:
Add manager integration test

gijskoning committed:
Add manager integration test

gijskoning committed:
Add manager integration test

gijskoning committed:
add todo

gijskoning committed:
add todo

gijskoning commented on common/text_window.py:
```diff
@@ -31,7 +32,8 @@ def close(self):
     if self.text_proc is not None:
-      while True:
+      params = Params()
+      while not params.get_bool('DoShutdown'):
```
This could halt during an exception

gijskoning replied:
Unsure about this change. Since the CI seemed to crash on the import line.
I moved it to the function and CI is passing, but I think this is just because it isn't executed in any test

gijskoning committed:
Fix

gijskoning committed:
Fix

gijskoning committed:
Fix test running sequentially

gijskoning committed:
Fix test running sequentially

gijskoning committed:
Fix Params import

gijskoning committed:
Fix Params import

gijskoning committed:
Fix Params import

gijskoning committed:
try fix

gijskoning committed:
try fix

gijskoning committed:
Remove change in text_window

gijskoning committed:
Remove change in text_window

gijskoning committed:
Remove change in text_window

gijskoning commented on tools/sim/test/test_carla_integration.py:
```diff
@@ -0,0 +1,108 @@
+    for p in expected_p:
+      name = managed_processes[p].get_process_state_msg().name
+      for proc in psutil.process_iter():
+        if proc.name().endswith(name):
```
@adeebshihadeh This assumes the first argument of the processes in selfdrive/manager/process_config.py
are also used at the end of process name. However just noticing that you are breaking that pattern here: https://github.com/commaai/openpilot/pull/24212
I probably need to change the classes in process_config such that they use the same attribute to define the name of the process.

gijskoning committed:
small cleanup

gijskoning committed:
small cleanup

gijskoning committed:
Merge remote-tracking branch 'gijs-openpilot/carla-bridge-ci' into carla-bridge-ci

gijskoning committed:
Merge branch 'master' into carla-wide-camera

# Conflicts:
#	tools/sim/bridge.py

gijskoning committed:
Remove c2 arg

gijskoning committed:
Update to 163 fov to not see through car

gijskoning committed:
update comment

gijskoning committed:
Merge branch 'carla-wide-camera' into carla-bridge-ci

# Conflicts:
#	tools/sim/bridge.py

gijskoning committed:
Fix test and check car engages

gijskoning committed:
commit

gijskoning committed:
Improve speed simulator by setting postprocess_effects to false.

gijskoning committed:
Add unblock stdout

pd0wm commented on tools/sim/bridge.py:
```diff
@@ -231,266 +234,289 @@ def can_function_runner(vs: VehicleState, exit_event: threading.Event):
+      blueprint.set_attribute('image_size_x', str(W))
+      blueprint.set_attribute('image_size_y', str(H))
+      blueprint.set_attribute('fov', str(fov))
+      blueprint.set_attribute('enable_postprocess_effects', 'False')
```
Gate this behind the `low-quality` flag and use that option during the test?
